distributed/tests/test_events.py

```diff
@@ -0,0 +1,220 @@
+import pickle
+
+from distributed import Event
+from distributed.utils_test import gen_cluster
+from distributed.utils_test import client, cluster_fixture, loop  # noqa F401
+
+
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 8)] * 2)
+async def test_event_on_workers(c, s, a, b):
+    # Test the "typical" use case of events:
+    # workers set, clear and wait for it
+    def wait_for_it_failing(x):
+        event = Event("x")
+
+        # Event is not set in another task so far
+        assert not event.wait(timeout=0.05)
+        assert not event.is_set()
+
+    def wait_for_it_ok(x):
+        event = Event("x")
+
+        # Event is set in another task
+        assert event.wait(timeout=0.5)
+        assert event.is_set()
+
+    def set_it():
+        event = Event("x")
+        event.set()
+
+    def clear_it():
+        event = Event("x")
+        event.clear()
+
+    wait_futures = c.map(wait_for_it_failing, range(10))
+    await c.gather(wait_futures)
+
+    set_future = c.submit(set_it)
+    await c.gather(set_future)
+
+    wait_futures = c.map(wait_for_it_ok, range(10))
+    await c.gather(wait_futures)
+
+    clear_future = c.submit(clear_it)
+    await c.gather(clear_future)
+
+    wait_futures = c.map(wait_for_it_ok, range(10))
+    await c.gather(wait_futures)
+
+    assert not s.extensions["events"]._events
+    assert not s.extensions["events"]._waiter_count
+
+
+@gen_cluster()
+async def test_default_event(s, a, b):
```

Comment:
```suggestion
@gen_cluster(client=True)
async def test_default_event(c, s, a, b):
```

I recommend including the client in these tests, otherwise it's a bit confusing about which client the `Event` is using by default.

In this case I suspect that this is only working because there is a worker in the same process, so the `get_worker()` path is saving us.  In principle though we shouldn't depend on this.  I recommend adding clients to most of the async tests here.

Reply:
Sounds very reasonable, thanks.